book/src/dev/rfcs/0011-value-pool.md

```diff
@@ -0,0 +1,229 @@
+- [Miner fees](https://github.com/ZcashFoundation/zebra/blob/62b3344f4d8605aad03133b8c0fa94cfb622d4a7/book/src/dev/rfcs/xxxx-block-subsidy.md#transaction-fees-calculation) - Fees collected for a block is the value pool for that block and they belong to the miner reward. 
+
+In addition, Zcashd has RPC calls that allow to get each pool on each block with `getblock()` and the value pools at current tip with `getblockchaininfo()`.
+
+# Definitions
+[definitions]: #definitions
+
+- `transparent pool` - Sum of all `tx_in` fields for transactions in the block chain, minus the sum of all `tx_out` fields for transactions in the block chain.
+- `sprout pool` - Sum of all `vpub_old` fields for transactions in the block chain, minus the sum of all `vpub_new` fields for transactions in the block chain [ZIP-209](https://zips.z.cash/zip-0209#terminology).
+- `sapling pool` - Negation of the sum of all `valueBalanceSapling` fields for transactions in the block chain [ZIP-209](https://zips.z.cash/zip-0209#terminology).
+- `orchard pool` - Negation of the sum of all `valueBalanceOrchard` fields for transactions in the block chain.
+- `transaction pool`
+- `block pool`
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Each of the 4 value pools change its value with every upcoming block. This is a state feature and Zebra handle this in the `zebra-state` crate. We propose to store the pool values among a reference block height into the disk.
+
+The proposed design goes from calculating transaction pools, summing them all to form block pools and finally summing all block pools to form the final pool value.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Changes to `zebra-chain/src/transaction.rs`
+
+We have a function to calculate each transaction pool for any transaction version:
+
+### `transparent_value_pool()`
+
+TODO
+
+### `sprout_value_pool()`
+
+```rust
+pub fn sprout_value_pool(&self) -> Amount {
+    match self {
+        Transaction::V1 { .. } => Amount::try_from(0).unwrap(),
+        Transaction::V2 { joinsplit_data, .. } |
+        Transaction::V3 { joinsplit_data, .. } => {
+            let mut value_balance: Amount<NegativeAllowed> = Amount::try_from(0).unwrap();
+            for js in joinsplit_data.as_ref().unwrap().joinsplits() {
+                value_balance = (value_balance + (js.vpub_old - js.vpub_new).unwrap().constrain()).unwrap();
+            }
+            value_balance
+        },
+        Transaction::V4 { joinsplit_data, .. } => {
+            let mut value_balance: Amount<NegativeAllowed> = Amount::try_from(0).unwrap();
+            for js in joinsplit_data.as_ref().unwrap().joinsplits() {
+                value_balance = (value_balance + (js.vpub_old - js.vpub_new).unwrap().constrain()).unwrap();
+            }
+            value_balance
+        },
```

Comment:
The code inside is the exact same however as `V2` and `V3` sprout data is `Bctv14Proof` and `V4` is `Groth16Proof`. This produce a build error when joined, i didn't researched much yet but it is probably an easy fix.